### ISSUE TITLE
Plugin crash

### ISSUE THREAD

**Issue by pannal** [jdf76/plugin.video.youtube#317]
This just happened:

```
04:25:34.171 T:3873436416   ERROR: GetDirectory - Error getting /home/osmc/.kodi/temp/plugin.video.youtube/
04:26:09.265 T:3702416128  NOTICE: [plugin.video.youtube] Running: YouTube (6.0.0~beta7) on Krypton (Kodi-17.6) with Python 2.7.13
04:26:15.248 T:3347051264  NOTICE: script.module.youtube.dl: youtube_dl core version: 2018.02.25
04:26:24.812 T:3347051264  NOTICE: script.module.youtube.dl: youtube_dl core version: 2018.02.25
04:59:46.923 T:3201766144  NOTICE: [plugin.video.youtube] Running: YouTube (6.0.0~beta7) on Krypton (Kodi-17.6) with Python 2.7.13
05:01:41.766 T:3201856256  NOTICE: [plugin.video.youtube] Running: YouTube (6.0.0~beta7) on Krypton (Kodi-17.6) with Python 2.7.13
05:02:00.940 T:3210244864  NOTICE: [plugin.video.youtube] Running: YouTube (6.0.0~beta7) on Krypton (Kodi-17.6) with Python 2.7.13
05:06:16.526 T:2864333568  NOTICE: [plugin.video.youtube] Running: YouTube (6.0.0~beta7) on Krypton (Kodi-17.6) with Python 2.7.13
15:23:24.114 T:2780300032  NOTICE: [plugin.video.youtube] Running: YouTube (6.0.0~beta7) on Krypton (Kodi-17.6) with Python 2.7.13
15:23:51.674 T:2788688640  NOTICE: [plugin.video.youtube] Running: YouTube (6.0.0~beta7) on Krypton (Kodi-17.6) with Python 2.7.13
15:23:53.609 T:4077621840   ERROR: GetDirectory - Error getting plugin://plugin.video.youtube/special/new_uploaded_videos_tv/
15:23:53.610 T:4077621840   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.youtube/special/new_uploaded_videos_tv/) failed
                                            Error Contents: [Errno 2] No such file or directory: '/home/osmc/.kodi/userdata/profiles/skintest2/addon_data/plugin.video.youtube/kodion/cache.sqlite'
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/default.py", line 7, in <module>
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/runner.py", line 41, in run
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_runner.py", line 24, in run
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/abstract_provider.py", line 107, in navigate
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/register_provider_path.py", line 11, in wrapper
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/provider.py", line 536, in _on_yt_specials
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/yt_specials.py", line 309, in process
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/yt_specials.py", line 273, in _process_new_uploaded_videos_tv
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/tv.py", line 46, in my_subscriptions_to_items
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/utils.py", line 439, in update_fanarts
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/resource_manager.py", line 166, in get_fanarts
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/resource_manager.py", line 50, in _update_channels
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/utils/function_cache.py", line 89, in get
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/utils/function_cache.py", line 52, in _get_cached_data
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/utils/storage.py", line 176, in _get
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/utils/storage.py", line 38, in _open
                                              File "/home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/utils/storage.py", line 96, in _optimize_file_size
                                            OSError: [Errno 2] No such file or directory: '/home/osmc/.kodi/userdata/profiles/skintest2/addon_data/plugin.video.youtube/kodion/cache.sqlite'
15:24:28.680 T:2763522816   ERROR: CPythonInvoker(58, /home/osmc/.kodi/addons/plugin.video.youtube/resources/lib/default.py): script didn't stop in 5 seconds - let's kill it
15:24:28.720 T:2813854464   ERROR: GetDirectory - Error getting plugin://plugin.video.youtube/special/new_uploaded_videos_tv/?reload=2018032314012720180323141844

```

**Comment by anxdpanic** [comment]
without the rest of the log I have no context on how that happened. does restarting kodi and the add-on resolve it?

**Comment by pannal** [comment]
I've updated the paste above with all the context I could find. I'll see whether this error stays.

**Comment by anxdpanic** [comment]
Strangest part for that error, is there there are 2 checks that the file exists before this error happens. I'll catch this error in the next bump though.

**Comment by pannal** [comment]
After a restart this didn't happen again.

**Comment by anxdpanic** [comment]
Closed by #318 8ba2ec83fc3bc12d4eb0f9c08f7c48e58be16534

**Comment by pannal** [comment]
When this happened I looked at the cache file and it was 0 bytes in size.

That's why https://github.com/jdf76/plugin.video.youtube/commit/8ba2ec83fc3bc12d4eb0f9c08f7c48e58be16534#diff-b0bb47df7a6ffff4d4d28a3f1223e3b9L94 didn't hit I guess.
